From 332d7649fb0f2db239af34f4cc83a8a636fccbec Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Thu, 10 Mar 2022 16:49:08 -0300
Subject: [PATCH 025/163] support: Fix support_process_state_wait path size
 calculation

Use INT_STRLEN_BOUND to proper get the maximum pid_t size.  Also
fix the wrong calculation (the 3 should multiply the sizeof (pid_t)).

Checked on x86_64-linux-gnu.

(cherry picked from commit 0374b487a71ad07005a816191dca829a7e6cdcae)
---
 support/support_process_state.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/support/support_process_state.c b/support/support_process_state.c
index 0dc608b468..ca86dc11e6 100644
--- a/support/support_process_state.c
+++ b/support/support_process_state.c
@@ -21,6 +21,7 @@
 #include <string.h>
 
 #include <array_length.h>
+#include <intprops.h>
 
 #include <support/process_state.h>
 #include <support/xstdio.h>
@@ -49,7 +50,7 @@ support_process_state_wait (pid_t pid, enum support_process_state state)
     { support_process_state_parked,       'P' },
   };
 
-  char spath[sizeof ("/proc/" + 3) * sizeof (pid_t) + sizeof ("/status") + 1];
+  char spath[sizeof ("/proc/") + INT_STRLEN_BOUND (pid_t) + sizeof ("/status") + 1];
   snprintf (spath, sizeof (spath), "/proc/%i/status", pid);
 
   FILE *fstatus = xfopen (spath, "r");
-- 
2.37.2

